applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/prebuckling_strategy.hpp

```diff
@@ -0,0 +1,980 @@
+    {
+        // Clear() controls order of deallocation to avoid invalid memory access
+        // in some special cases.
+        // warning: BaseType::GetModelPart() may be invalid here.
+        this->Clear();
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void SetIsInitialized(bool val)
+    {
+        mInitializeWasPerformed = val;
+    }
+
+    bool GetIsInitialized() const
+    {
+        return mInitializeWasPerformed;
+    }
+
+    SchemePointerType &pGetScheme()
+    {
+        return mpScheme;
+    };
+
+    BuilderAndSolverPointerType &pGetEigenSolver()
+    {
+        return mpEigenSolver;
+    };
+
+    BuilderAndSolverPointerType &pGetBuilderAndSolver()
+    {
+        return mpBuilderAndSolver;
+    };
+
+    // SparseMatrixPointerType &pGetMassMatrix()
+    // {
+    //     return mpMassMatrix;
+    // }
+
+    SparseMatrixPointerType &pGetStiffnessMatrixPrevious()
+    {
+        return mpStiffnessMatrixPrevious;
+    }
+
+    SparseMatrixPointerType &pGetStiffnessMatrix()
+    {
+        return mpStiffnessMatrix;
+    }
+
+    SparseVectorPointerType &pGetRHS()
+    {
+        return mpRHS;
+    }
+
+    SparseVectorPointerType &pGetDx()
+    {
+        return mpDx;
+    }
+
+    // SparseMatrixType &GetMassMatrix()
+    // {
+    //     return *mpMassMatrix;
+    // }
+
+    SparseMatrixType &GetStiffnessMatrixPrevious()
+    {
+        return *mpStiffnessMatrixPrevious;
+    }
+
+    SparseMatrixType &GetStiffnessMatrix()
+    {
+        return *mpStiffnessMatrix;
+    }
+
+    SparseVectorType &GetRHS()
+    {
+        return *mpRHS;
+    }
+
+    SparseVectorType &GetDx()
+    {
+        return *mpDx;
+    }
+
+    ConvergenceCriteriaType &GetConvergenceCriteria()
+    {
+        return mpConvergenceCriteria;
+    }
+
+    bool GetSolutionFoundFlag()
+    {
+        return mSolutionFound;
+    }
+
+    void SetReformDofSetAtEachStepFlag(bool flag)
+    {
+        this->pBuilderAndSolver()->SetReshapeMatrixFlag(flag);
+    }
+
+    bool GetReformDofSetAtEachStepFlag() const
+    {
+        return this->pBuilderAndSolver()->GetReshapeMatrixFlag();
+    }
+
+    /// Set verbosity level of the solving strategy.
+    /**
+     * - 0 -> mute... no echo at all
+     * - 1 -> print time and basic information
+     * - 2 -> print linear solver data
+     * - 3 -> print debug information
+     */
+    void SetEchoLevel(int Level) override
+    {
+        BaseType::SetEchoLevel(Level);
+        this->pGetEigenSolver()->SetEchoLevel(Level);
+        this->pGetBuilderAndSolver()->SetEchoLevel(Level);
+    }
+
+    /**
+     * Initialization to be performed once before using the strategy.
+     */
+    void Initialize() override
+    {
+        KRATOS_TRY
+
+        ModelPart &rModelPart = BaseType::GetModelPart();
+        const int rank = rModelPart.GetCommunicator().MyPID();
+        typename ConvergenceCriteriaType::Pointer pConvergenceCriteria = mpConvergenceCriteria;
```

Comment:
why in this functions you create all those aux-variables

Reply:
I removed them.